extensions/cloud/portability-cloud-google/src/main/java/org/datatransferproject/cloud/google/GoogleJobStore.java

```diff
@@ -327,10 +300,62 @@ private Key getDataKey(UUID jobId, String key) {
     if (entity == null) {
       return null;
     }
+    String serializedEntity = entity.getString(type.getName());
     try {
-      return objectMapper.readValue(entity.getString(type.getName()), type);
+      return objectMapper.readValue(serializedEntity, type);
     } catch (IOException t) {
-      throw new RuntimeException("Failed to deserialized entityKey: " + entityKey, t);
+      throw new RuntimeException("Failed to deserialize entity: " + serializedEntity, t);
     }
   }
+
+  @Override
+  public void create(UUID jobId, String key, InputStream stream) {
+    // TODO: use result to determine success/failure
+    googleUserDataStore.create(jobId, key, stream);
+  }
+
+  @Override
+  public InputStream getStream(UUID jobId, String key) {
+    com.google.cloud.storage.Blob blob = googleUserDataStore.get(jobId, key);
```

Comment:
this is still blob logic within GoogleJobStore, right? 

can all the blob logic be moved into UserDataStore?